clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteConfigurationBuilder.java

```diff
@@ -0,0 +1,177 @@
+import org.infinispan.client.hotrod.configuration.ExecutorFactoryConfiguration;
+import org.infinispan.client.hotrod.configuration.NearCacheConfiguration;
+import org.jboss.as.clustering.controller.CommonUnaryRequirement;
+import org.jboss.as.clustering.controller.ResourceServiceBuilder;
+import org.jboss.as.clustering.infinispan.InfinispanLogger;
+import org.jboss.as.clustering.infinispan.subsystem.ThreadPoolResourceDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.StringListAttributeDefinition;
+import org.jboss.as.network.OutboundSocketBinding;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.Property;
+import org.jboss.msc.service.ServiceBuilder;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.service.ServiceTarget;
+import org.jboss.msc.service.ValueService;
+import org.jboss.msc.value.InjectedValue;
+import org.jboss.msc.value.Value;
+import org.wildfly.clustering.service.Builder;
+import org.wildfly.clustering.service.InjectedValueDependency;
+import org.wildfly.clustering.service.ValueDependency;
+
+/**
+ * @author Radoslav Husar
+ */
+public class RemoteConfigurationBuilder implements ResourceServiceBuilder<Configuration>, Value<Configuration> {
+
+    private final PathAddress address;
+
+    private final Map<String, List<InjectedValueDependency<OutboundSocketBinding>>> clusters = new HashMap<>();
+    private final InjectedValue<ConnectionPoolConfiguration> connectionPool = new InjectedValue<>();
+    private final InjectedValue<NearCacheConfiguration> nearCache = new InjectedValue<>();
+    private final Map<ThreadPoolResourceDefinition, ValueDependency<ExecutorFactoryConfiguration>> threadPools = new EnumMap<>(ThreadPoolResourceDefinition.class);
+
+    private volatile int connectionTimeout;
+    private volatile String defaultRemoteCluster;
+    private volatile int keySizeEstimate;
+    private volatile int maxRetries;
+    private volatile String protocolVersion;
+    private volatile int socketTimeout;
+    private volatile boolean tcpNoDelay;
+    private volatile boolean tcpKeepAlive;
+    private volatile int valueSizeEstimate;
+
+    RemoteConfigurationBuilder(PathAddress address) {
+        this.address = address;
+        EnumSet.of(ThreadPoolResourceDefinition.CLIENT).forEach(pool -> this.threadPools.put(pool, new InjectedValueDependency<>(pool.getServiceName(address), ExecutorFactoryConfiguration.class)));
+    }
+
+    @Override
+    public ServiceName getServiceName() {
+        return CONFIGURATION.getServiceName(this.address);
+    }
+
+    @Override
+    public Builder<Configuration> configure(OperationContext context, ModelNode model) throws OperationFailedException {
+        this.connectionTimeout = Attribute.CONNECTION_TIMEOUT.getDefinition().resolveModelAttribute(context, model).asInt();
```

Comment:
You don't need to call getDefinition() - this method is exposed via the Attribute interface.

Reply:
Done, removed in all other places.